frontend/src/StoryDisplays/Stories.jsx

```diff
@@ -1,25 +1,147 @@
-
-    useEffect(() => {
-        fetch('http://localhost:3000/api/stories')
-            .then(response => response.json())
-            .then(data => setStories(data))
-            .catch(error => console.error('Error fetching stories:', error));
-    }, []);
-
-    return (
-        <div className="stories-container">
-            {stories.map(story => (
-                <div key={story.id} className="story-circle">
-                    <img src={story.fileUrl} alt="Story" className="story-image" />
-                </div>
-            ))}
+const Stories = ({ currentUser }) => {
+  const [stories, setStories] = useState([]);
+  const storiesListRef = useRef(new StoriesList());
+  const [viewingStories, setViewingStories] = useState(null);
+
+  const fetchStories = useCallback(() => {
+    fetch('http://localhost:3000/api/stories')
+      .then(response => response.json())
+      .then(data => {
+        const storiesList = storiesListRef.current;
+        storiesList.removeExpiredStories();
+        data.forEach(userStories => {
+          if (userStories && userStories.stories) {
+            userStories.stories.forEach(story => {
+              if (story) {
+                storiesList.insertStory(story);
+              }
+            });
+          }
+        });
+        setStories(storiesList.toArray());
+      })
+      .catch(error => console.error('Error fetching stories:', error));
+  }, []);
+
+  useEffect(() => {
+    fetchStories();
+    const interval = setInterval(fetchStories, 60000);
+    return () => clearInterval(interval);
+  }, [fetchStories]);
+
+  const addNewStory = async (file, userId) => {
+    const formData = new FormData();
+    formData.append('story', file);
+    formData.append('userId', userId);
+    try {
+      const response = await fetch('http://localhost:3000/api/stories', {
+        method: 'POST',
+        body: formData,
+      });
+      if (response.ok) {
+        const newStory = await response.json();
+        await fetchStories(); // Fetch the updated list of stories
+      } else {
+        console.error('Failed to upload story');
+      }
+    } catch (error) {
+      console.error('Error uploading story:', error);
+    }
+  };
+
+  const removeExpiredStories = useCallback(async () => {
+    try {
+      const response = await fetch('http://localhost:3000/api/stories/expired', {
+        method: 'DELETE',
+      });
+      if (response.ok) {
+        storiesListRef.current.removeExpiredStories();
+        setStories(storiesListRef.current.toArray());
+      } else {
+        console.error('Failed to remove expired stories');
+      }
+    } catch (error) {
+      console.error('Error removing expired stories:', error);
+    }
+  }, []);
+
+  useEffect(() => {
+    const interval = setInterval(removeExpiredStories, 3600000);
+    return () => clearInterval(interval);
+  }, [removeExpiredStories]);
+
+  const groupStoriesByUser = (stories) => {
+    return stories.reduce((acc, story) => {
```

Comment:
It's great to see you consistently making use of functions like reduce to simplify logic!